src/main.rs

```diff
@@ -1,40 +1,40 @@
-#[macro_use]
-extern crate serde_json;
-
-#[macro_use]
-extern crate bson;
-
-#[macro_use]
-extern crate log;
-
-use actix_files as fs;
-use actix_web::{web, App, HttpServer};
-use handlebars::*;
+use crate::search::indexing::index_mods;
 use actix_web::middleware::Logger;
+use actix_web::{web, App, HttpServer};
 use env_logger::Env;
 use std::env;
+use std::fs::File;
+use log::info;
 
 mod database;
-mod helpers;
+mod models;
 mod routes;
+mod search;
 
 #[actix_rt::main]
 async fn main() -> std::io::Result<()> {
     env_logger::from_env(Env::default().default_filter_or("info")).init();
     dotenv::dotenv().ok();
 
-    //Handlebars
-    let mut handlebars = Handlebars::new();
-
-    helpers::register_helpers(&mut handlebars);
-    handlebars
-        .register_templates_directory(".hbs", "./templates")
-        .unwrap();
-
-    let handlebars_ref = web::Data::new(handlebars);
-
     let client = database::connect().await.unwrap();
-    routes::index_mods(client).await.unwrap();
+
+    // Get executable path
+    let mut exe_path = env::current_exe()?.parent().unwrap().to_path_buf();
+    // Create the path to the index lock file
+    exe_path.push("index.v1.lock");
+
+    //Indexing mods if not already done
+    if env::args().any(|x| x == "regen") {
```

Comment:
Although we may not want to pull in a heavy dependency (ie. `clap` or `structopt`), we may want a real argument parsing library to handle this.  We may want to handle that after this PR, though.

Reply:
This solution was temporary